crates/chainspec/src/spec.rs

```diff
@@ -1085,32 +1088,21 @@ impl DepositContract {
 #[derive(Default, Debug, serde::Deserialize)]
 #[serde(rename_all = "camelCase")]
 struct OptimismGenesisInfo {
-    bedrock_block: Option<u64>,
-    regolith_time: Option<u64>,
-    canyon_time: Option<u64>,
-    ecotone_time: Option<u64>,
-    fjord_time: Option<u64>,
+    genesis_info: OpGenesisInfo,
     #[serde(skip)]
     base_fee_params: BaseFeeParamsKind,
 }
 
-#[cfg(feature = "optimism")]
-#[derive(Debug, Eq, PartialEq, serde::Deserialize)]
-#[serde(rename_all = "camelCase")]
-struct OptimismBaseFeeInfo {
-    eip1559_elasticity: Option<u64>,
-    eip1559_denominator: Option<u64>,
-    eip1559_denominator_canyon: Option<u64>,
-}
-
 #[cfg(feature = "optimism")]
 impl OptimismGenesisInfo {
     fn extract_from(genesis: &Genesis) -> Self {
-        let mut optimism_genesis_info: Self =
-            genesis.config.extra_fields.deserialize_as().unwrap_or_default();
+        let mut optimism_genesis_info = OptimismGenesisInfo::default();
+        if let Some(genesis_info) = OpGenesisInfo::extract_from(&genesis.config.extra_fields) {
+            optimism_genesis_info.genesis_info = genesis_info;
+        }
 
-        if let Some(Ok(optimism_base_fee_info)) =
-            genesis.config.extra_fields.get_deserialized::<OptimismBaseFeeInfo>("optimism")
+        if let Some(optimism_base_fee_info) =
+            OptimismBaseFeeInfo::extract_from(&genesis.config.extra_fields)
```

Comment:
this can be collapsed into `OptimismChainInfo::extract_from`

Reply:
fixed. thx